### ISSUE TITLE
Source code missing in howto.hellodockernginx for mysql

### ISSUE THREAD

**Issue by hendrikdutoit** [reahl/reahl#406]
The howto.hellodockernginx example does not work.  If you checkout the example with 

```
reahl example howtos.hellodockernginx
```
it seems that some of the source code is missing:
`prod` folder and files
`scripts` folder and files
`www ` folder and files
`dockerignore` file

It also seems that the `.reahlproject` file is missing.

**Comment by IwanVosloo** [comment]
Thanks for the report @hendrikdutoit 

Is this on Reahl 6.1 ? 
On which OS platform did you get this problem?

Assuming a Reahl 6.0 or 6.1, I can answer that there is no longer a .reahlproject, we have switched in this version to having a setup.cfg + pyproject.toml, see: https://www.reahl.org/docs/6.0/whatchanged.d.html#doing-away-with-reahlproject

The missing source we'll have to reproduce to see what's wrong there.

**Comment by hendrikdutoit** [comment]
# Stack
Reahl 6.1
Windows 10
Python 3.11

# .reahlproject
I copied the missing folders into the hellodockernginx project folder.  During the creation of the Docker image, it complained about a missing .reahlproject file.  However, I am not sure if this issue is due to the missing files or if these are two separate issues.

```
652.7 Installing collected packages: webencodings, wcwidth, sortedcontainers, rjsmin, reahl, rcssmin, pytz, ply, zipp, wrapt, WebOb, watchdog, urllib3, tzdata, toml, soupsieve, sniffio, six, rfc3986, reahl-stubble, PySocks, Pygments, pycparser, prompt-toolkit, pkginfo, more-itertools, mdurl, MarkupSafe, lxml, jeepney, idna, h11, greenlet, exceptiongroup, docutils, charset-normalizer, certifi, Babel, attrs, wsproto, SQLAlchemy, requests, pytz-deprecation-shim, python-dateutil, outcome, markdown-it-py, Mako, jaraco.classes, importlib-metadata, cffi, bleach, beautifulsoup4, tzlocal, trio, rich, requests-toolbelt, reahl-component, readme-renderer, cryptography, alembic, trio-websocket, SecretStorage, reahl-tofu, reahl-mailutil, reahl-commands, selenium, reahl-web, reahl-sqlalchemysupport, keyring, twine, reahl-dev, reahl-webdev
657.3 Successfully installed Babel-2.11.0 Mako-1.2.4 MarkupSafe-2.1.3 PySocks-1.7.1 Pygments-2.13.0 SQLAlchemy-1.4.49 SecretStorage-3.3.3 WebOb-1.8.7 alembic-1.8.1 attrs-23.1.0 beautifulsoup4-4.11.2 bleach-6.0.0 certifi-2023.7.22 cffi-1.15.1 charset-normalizer-3.2.0 cryptography-41.0.3 docutils-0.19 exceptiongroup-1.1.3 greenlet-2.0.2 h11-0.14.0 idna-3.4 importlib-metadata-6.8.0 jaraco.classes-3.3.0 jeepney-0.8.0 keyring-24.2.0 lxml-4.9.3 markdown-it-py-3.0.0 mdurl-0.1.2 more-itertools-10.1.0 outcome-1.2.0 pkginfo-1.9.6 ply-3.11 prompt-toolkit-3.0.39 pycparser-2.21 python-dateutil-2.8.2 pytz-2023.3 pytz-deprecation-shim-0.1.0.post0 rcssmin-1.1.1 readme-renderer-41.0 reahl-6.1.1 reahl-commands-6.1.1 reahl-component-6.1.1 reahl-dev-6.1.1 reahl-mailutil-6.1.1 reahl-sqlalchemysupport-6.1.1 reahl-stubble-6.1.1 reahl-tofu-6.1.1 reahl-web-6.1.1 reahl-webdev-6.1.1 requests-2.31.0 requests-toolbelt-1.0.0 rfc3986-2.0.0 rich-13.5.2 rjsmin-1.2.1 selenium-4.7.2 six-1.16.0 sniffio-1.3.0 sortedcontainers-2.4.0 soupsieve-2.4.1 toml-0.10.2 trio-0.22.2 trio-websocket-0.10.3 twine-4.0.2 tzdata-2023.3 tzlocal-4.2 urllib3-1.26.16 watchdog-2.2.1 wcwidth-0.2.6 webencodings-0.5.1 wrapt-1.14.1 wsproto-1.2.0 zipp-3.16.2
658.1 Traceback (most recent call last):
658.1   File "/tmp/build-venv/bin/reahl", line 8, in <module>
658.1     sys.exit(ReahlCommandline.execute_one())
658.1   File "/tmp/build-venv/lib/python3.10/site-packages/reahl/component/shelltools.py", line 206, in execute_one
658.1     result = cls().do(sys.argv[1:])
658.1   File "/tmp/build-venv/lib/python3.10/site-packages/reahl/component/shelltools.py", line 111, in do
658.1     return self.execute(args)
658.1   File "/tmp/build-venv/lib/python3.10/site-packages/reahl/component/shelltools.py", line 226, in execute
658.1     return super().execute(args)
658.1   File "/tmp/build-venv/lib/python3.10/site-packages/reahl/component/shelltools.py", line 170, in execute
658.1     return command.do(args.command_args)
658.1   File "/tmp/build-venv/lib/python3.10/site-packages/reahl/component/shelltools.py", line 111, in do
658.1     return self.execute(args)
658.1   File "/tmp/build-venv/lib/python3.10/site-packages/reahl/dev/devshell.py", line 275, in execute
658.1     results[i] = self.execute_one(i, args)
658.1   File "/tmp/build-venv/lib/python3.10/site-packages/reahl/dev/devshell.py", line 217, in execute_one
658.1     retcode = self.function(project, args)
658.1   File "/tmp/build-venv/lib/python3.10/site-packages/reahl/dev/devshell.py", line 415, in function
658.1     project.build(sign=self.sign)
658.1   File "/tmp/build-venv/lib/python3.10/site-packages/reahl/dev/devdomain.py", line 1719, in build
658.1     assert self.packages_to_distribute, 'For %s: No <package>... listed in .reahlproject, nothing to do.' % self.project_name
658.1 AssertionError: For hellodockernginx: No <package>... listed in .reahlproject, nothing to do.
------
failed to solve: process "/bin/sh -c apt-get update -y && apt-get install -y --no-install-recommends python3-pip python3-venv $DB_CLIENT_BUILD_LIBS &&     apt-get clean && rm -rf /var/cache/apt/* &&  rm -rf /var/lib/apt/lists/* &&     python3 -m venv $VENV &&     $VENV/bin/pip install -U pip &&     $VENV/bin/pip install wheel &&     if [ ! -z \"$WHEELS_TO_BUILD\" ]; then $VENV/bin/pip wheel -w /tmp/wheels $WHEELS_TO_BUILD; else mkdir /tmp/wheels ;fi &&     $VENV/bin/pip install reahl[dev]$REAHL_VERSION_SPEC &&     $VENV/bin/reahl build -ns &&     mv /app/.reahlworkspace/dist-egg/* /tmp/wheels/" did not complete successfully: exit code: 1

```

**Comment by IwanVosloo** [comment]
Ah, I think the error message is misleading and needs to be updated. I think the setup.cfg is probably also missing from the checked-out source.

Thanks, we'll have to try this on windows.

**Comment by craig-reahl** [comment]
I can **confirm** the issue being a **packaging** issue of reahl-doc version 6.1.1. Using the example checkout command of

`reahl example howtos.hellodockernginx`

it should yield

```
hellodockernginx/
├── docker-compose.yml
├── etc
│   └── web.config.py
├── hellodockernginx.py
├── hellodockernginx.wsgi.py
├── prod
│   ├── Dockerfile
│   ├── etc
│   │   ├── mailutil.config.py
│   │   ├── reahl.config.py
│   │   ├── systemaccountmodel.config.py
│   │   └── web.config.py
│   ├── nginx
│   │   ├── Dockerfile
│   │   └── app.conf
│   └── uwsgi
│       └── app.ini
├── pyproject.toml
├── setup.cfg
├── scripts
│   └── setup_database.sh
└── www
```

Sadly, in the [package wheel file on pypi ](https://files.pythonhosted.org/packages/0b/44/ff0e170f9afa31b493b34fc586dda20bae395f977effcb692ece5a0d2228/reahl_doc-6.1.1-py3-none-any.whl) for [https://pypi.org/project/reahl-doc](https://pypi.org/project/reahl-doc)

it only contains this subset of files:

```
hellodockernginx/
├── etc
│   └── web.config.py
├── hellodockernginx.py
├── hellodockernginxwsgi.py
├── __init__.py
├── pyproject.toml
└── setup.cfg
```

If you need the source code that was omitted in the 6.1.1 release, you could get it from here:
[https://github.com/reahl/reahl/tree/6.1.1/reahl-doc/reahl/doc/examples/howtos/hellodockernginx](https://github.com/reahl/reahl/tree/6.1.1/reahl-doc/reahl/doc/examples/howtos/hellodockernginx)

**Comment by IwanVosloo** [comment]
Fixed in #404
